packages/devtools_app/lib/src/timeline/flutter/flutter_frames_chart.dart

```diff
@@ -376,21 +375,15 @@ class SelectedDataPoint extends LineChartMarker {
     );
     painter.paint(canvas, pos);
     canvas.restore();
-
-    if (onSelected != null && _lastFrameIndex != frameIndex) {
-      // Only fire when a different frame is selected.
-      // TODO(terry): reconfigure this code as selection should not be happening
-      // during paint. This task scheduling is a hack.
-      SchedulerBinding.instance.scheduleTask(
-        () => onSelected(frameIndex),
-        Priority.animation,
-      );
-      _lastFrameIndex = frameIndex;
-    }
   }
 
   @override
-  void refreshContent(Entry e, Highlight highlight) {
+  void refreshContent(Entry e, Highlight highlight) async {
     _entry = e;
+    final frameIndex = _entry.x.toInt();
```

Comment:
is x a double or is it just typed as a num?

Reply:
if it is really an int, probably better to wrote
final int frameIndex = _entry.x;

Reply:
x is a double.